src/tarantool-gdb.py

```diff
@@ -0,0 +1,1371 @@
+                k -= 1
+                continue
+            elif l > cls.MP_HINT:
+                k -= l;
+                k -= 1
+                continue
+            else:
+                data.seek(-1)
+                cls.next_slowpath(data, k)
+                return
+
+    def to_string_data(cls, data, depth):
+        s = str()
+        mp_type = cls.typeof(data)
+
+        if mp_type == cls.MP_NIL:
+            cls.decode_nil(data)
+            s += 'null'
+
+        elif mp_type == cls.MP_UINT:
+            s += str(cls.decode_uint(data)) + 'U'
+
+        elif mp_type == cls.MP_INT:
+            s += str(cls.decode_int(data))
+
+        elif mp_type == cls.MP_STR or mp_type == cls.MP_BIN:
+            len = cls.decode_strl(data) if mp_type == cls.MP_STR else cls.decode_binl(data)
+            s += '"'
+            for i in range(0, len):
+                c = data.read_u8()
+                if c < 128 and cls.mp_char2escape[c] != 0:
+                    s += cls.mp_char2escape[c].string()
+                else:
+                    s += chr(c)
+            s += '"'
+
+        elif mp_type == cls.MP_ARRAY:
+            s += '['
+            if depth == 0:
+                s += '...'
+                cls.next(data)
+            else:
+                depth -= 1
+                count = cls.decode_array(data)
+                for i in range(0, count):
+                    if i > 0:
+                        s += ', '
+                    s += cls.to_string_data(data, depth)
+                depth += 1
+            s += ']'
+
+        elif mp_type == cls.MP_MAP:
+            s += '{'
+            if depth == 0:
+                s += '...'
+                cls.next(data)
+            else:
+                depth -= 1
+                count = cls.decode_map(data)
+                for i in range(0, count):
+                    if i > 0:
+                        s += ', '
+                    s += cls.to_string_data(data, depth)
+                    s += ': '
+                    s += cls.to_string_data(data, depth)
+                depth += 1
+            s += '}'
+
+        elif mp_type == cls.MP_BOOL:
+            s += 'true' if cls.decode_bool(data) else 'false'
+
+        elif mp_type == cls.MP_FLOAT:
+            s += str(cls.decode_float(data))
+
+        elif mp_type == cls.MP_DOUBLE:
+            s += str(cls.decode_double(data))
+
+        elif mp_type == cls.MP_EXT:
+            ext_len, ext_type = cls.decode_extl(data)
+            s += cls.to_string_ext(data, depth, ext_len, ext_type)
+
+        else:
+            raise gdb.GdbError('print_recursive: unexpected type ({})'.format(mp_type))
+
+        return s
+
+    def __init__(self, val):
+        self.val = val
+
+    def to_string(self, depth=-1, maxlen=-1):
+        s = self.to_string_data(InputStream(self.val), depth)
+        return s if maxlen < 0 else s[:maxlen]
+
+    def __str__(self):
+        return self.to_string()
+
+
+class TtMsgPack(MsgPack):
```

Comment:
I'm considering that semantically "find" means that the entity we are searching for might not exist. If so `find_value` just returns `None` and subsequent code deals with that.
In base class all "constants" must exist, and `parse_and_eval` will throw an exception so we will get an error in gdb console and it is correct, because the lack of these "constants" means that something went terribly wrong )